### ISSUE TITLE
SublimeLinter-eslint not working when eslint es installed locally

### ISSUE THREAD

**Issue by doberkofler** [roadhump/SublimeLinter-eslint#23]
I just noticed, that when eslint is also installed locally the following error is reported:
```
SublimeLinter: eslint: ods.ui.form.js ['D:\\MyDev\\ljs_app\\trunk\\periscope\\build\\node_modules\\.bin\\eslint', '--format=compact', '--stdin'] 
SublimeLinter: ERROR: could not launch ['D:\\MyDev\\ljs_app\\trunk\\periscope\\build\\node_modules\\.bin\\eslint', '--format=compact', '--stdin'] 
SublimeLinter: reason: [WinError 193] %1 is not a valid Win32 application 
```
I'm running ST3 Build 3065 on a Windows 7 x64 system and when eslint is only installed globally, the linter plugin works as expected. I'm using eslint 0.14.0.

I also noticed that (other than the jscs and jshint) plugins eslint is not shown when loading ST.
I just wanted to mention it but it actually should not be related to this problem as it is also not shown when eslint is only installed globally and the plugin is working:
```
SublimeLinter: jscs version query: C:\Users\doberkofler.LBITS\AppData\Roaming\npm\jscs.cmd --version 
SublimeLinter: jscs version: 1.10.0 
SublimeLinter: jscs: (>= 1.0.10) satisfied by 1.10.0 
SublimeLinter: jscs activated: C:\Users\doberkofler.LBITS\AppData\Roaming\npm\jscs.cmd 
SublimeLinter: jshint version query: C:\Users\doberkofler.LBITS\AppData\Roaming\npm\jshint.cmd --version 
SublimeLinter: jshint version: 2.6.0 
SublimeLinter: jshint: (>= 2.5.0) satisfied by 2.6.0 
SublimeLinter: jshint activated: C:\Users\doberkofler.LBITS\AppData\Roaming\npm\jshint.cmd 
SublimeLinter: found existing HTML syntax, version 3 
SublimeLinter: found existing Rails syntax, version 1 
```

**Comment by roadhump** [comment]
Hello!

Can you try to execute local eslint in cmd separately, copy command from ST console? It seems that error is pretty straightforward and it can be something wrong with your local eslint.

```
D:\\MyDev\\ljs_app\\trunk\\periscope\\build\\node_modules\\.bin\\eslint file.js
```

**Comment by doberkofler** [comment]
Thanks for the quick feedback. I've tried to call eslint with the appropriate path from the console but unfortunately from the command line it did work as expected but in ST3, the above error is shown.

**Comment by roadhump** [comment]
I strongly suppose there is a bug in SublimeLinter or/and Windows. SublimeLinter-eslint was changed to be a subclass of `NodeLinter` from SublimeLinter and it seems no other linters that did the same, what would help to compare your case with. I suggest you to post bug to SublimeLinter repo.

**Comment by dictions** [comment]
Bump. I'm having a similar issue, but doesn't seem to be the same cause. 

All I have to do is backup the `.bin` folder in `node_modules` (reverting to a globally installed eslint) and everything works like a charm. I'm not getting any errors in the Sublime console either. It says eslint has loaded but doesn't result in any errors (which it does if I'm using the globally installed eslint).

Thoughts?

**Comment by roadhump** [comment]
@dictions Try to install eslint locally, then restart ST, it helps often.

**Comment by 0x80** [comment]
@roadhump I have a similar problem. I removed my globally installed eslint and am now trying to get a locally installed version working but sublime can't find it. I've turned on debugging for the linter and this is the console output:

```
SublimeLinter: WARNING: eslint deactivated, cannot locate local or global binary 
SublimeLinter: ERROR: eslint cannot locate 'eslint' 
```

The linter is in my project root node_modules/.bin and working fine if I execute it. I've restarted sublime several times. What else can I do?

**Comment by roadhump** [comment]
@0x80 As I know there should be `package.json` and `eslint` installed as `npm install eslint --save-dev` (or `--save`).

**Comment by 0x80** [comment]
@roadhump AHA!! the package.json file was the missing piece. Is this documented somewhere? I have wasted a lot of time because of that and I suspect other people do to. Other posts from people that complain that sublime can't find their instance are usually answered with "install globally and point sublime to the absolute path" but that isn't good practice especially if you use NVM. Anyways I'm glad I finally get it.

Would it be possible for the linter to just look in ./node_modules/.bin instead of requiring a package.json file in the root? I don't see how any essential information is coming from that file.

**Comment by roadhump** [comment]
@0x80 I've add installation info to README, these questions are come too often. They should be covered in SublimeLinter docs, but they does not now and it seems like nobody will try to find them there.

**Comment by 0x80** [comment]
:+1:

**Comment by linxiaowu66** [comment]
it still can not work. I had tried all methods, it still can not find my local eslint instance. Can someone to solve this problem completely ? Thanks my friends....
